string_manipulation.rb

```diff
@@ -208,7 +218,10 @@ def reverse_sentence(my_sentence)
   return my_sentence
 end
 
+# Time complexity:  first while loop is O(n), which is reversing my_sentence (swapping) and the send while loop is O(n^2)
+# n + n2 = O(n^2).  It is O(n2) because there are 2 while loops on the same level; one while loop has two nested loops and the other while loop doesn't.  For each increase in put size, the speed of the algorithm will double.
```

Comment:
As explained in reverse_words, the second while loop will be of O(2n) and not O(n^2). Bringing the overall time complexity to O(3n) or simply O(n).
Aside: notice how you could simply call the first two methods one after the other to achieve what you've implemented here. :-)